packages/react/src/components/DesignLibrary.test.tsx

```diff
@@ -378,31 +385,49 @@ describe('<DesignLibrary />', () => {
     it('should render error message when component fails to initialize', async () => {
       const rendered = render(
         <SitecoreProvider componentMap={components} api={api} page={page}>
-          <DesignLibrary importMap={defaultImportMap} />
+          <DesignLibrary loadImportMap={defaultImportMap} />
         </SitecoreProvider>,
         { container: document.body }
       );
 
-      fireEvent('Error', null);
-
       await waitFor(() => {
+        callbackEvent('Error', null);
         expect(rendered.baseElement.innerHTML).to.contain('Error during component initialization');
       });
     });
 
-    it('should render error message when component fails to render', async () => {
+    it('should render error message when component fails to initialize due to failing import map promise', async () => {
+      const initError = new Error('Failed to load import map');
+      const errorImportMap = () =>
+        new Promise((_, reject) => {
+          reject(initError);
+        });
       const rendered = render(
         <SitecoreProvider componentMap={components} api={api} page={page}>
-          <DesignLibrary importMap={defaultImportMap} />
+          <DesignLibrary loadImportMap={errorImportMap} />
         </SitecoreProvider>,
         { container: document.body }
       );
 
-      fireEvent(null, () => {
-        throw new Error('Error rendering component');
+      await waitFor(() => {
+        callbackEvent('Error', null);
```

Comment:
I beliebe that here we need to pass not an error but real component, since callback shouldn't throw an error in this test case
```suggestion
        callbackEvent('Error', null);
```